includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1514,7 +1514,10 @@ public function process_payment_for_order( $cart, $payment_information, $schedul
 				if ( $payment_information->is_using_saved_payment_method() ) {
 					$billing_details = WC_Payments_Utils::get_billing_details_from_order( $order );
 
-					if ( ! empty( $billing_details ) ) {
+					$is_legacy_card_object = strpos( $payment_information->get_payment_method() ?? '', 'card_' ) === 0;
+
+					// Don't update billing details for legacy card objects because they are not supported.
```

Comment:
```suggestion
					// Not updating billing details for legacy card objects because they have a different structure and are no longer supported.
```

I was thinking of adding a note on the different structure - but then I had to limit the length of the comment 😅 let me know your thoughts if it's worth adding.

Reply:
Changed the wording, always better to be as explicit as possible 👍 Thanks!